src/app/asset-page/asset-page.component.ts

```diff
@@ -1787,6 +1790,46 @@ export class AssetPage implements OnInit, OnDestroy {
         this.angulartics.eventTrack.next({ properties: { event: 'quizModeTooltipDismissed', category: 'promotion', label: this.assetIds[0] } })
     }
 
+  /**
+   * when user clicks Accept on terms & conditions modal, sets flag and
+   * logs event
+   * @param: downloadType
+   */
+  public handleAcceptance(): void {
+    this.acceptedTerms = true
+
+    this.trackItemDownload(this.assets[0])
+  }
+
+  public userHasAcceptedTerms() {
+    let asset = this.assets[0]
+
+    return (asset.publicDownload || !asset.disableDownload) && this.acceptedTerms
+  }
+
+  private handleDownloadClick(downloadType): void {
+    this.downloadType = downloadType
+    let asset = this.assets[0]
+
+    if(this.userHasAcceptedTerms()) {
+      this.trackItemDownload(asset)
+    } else {
+      if(downloadType === "image") {
+        this.showAgreeModalForDownloadImage()
+      } else {
+        this.showAgreeModalForDownloadView()
+      }
+    }
+  }
+
+    public handleDownloadViewClick() {
```

Comment:
Nit - Indenting here looks a little off, but it's hard to tell in this view

Reply:
Ya the indenting in this file seems messed up. Some places are 4 spaces, while other is 2. I messed with my editor settings trying to make it 4, but apparently we hardcode the indent size in this file: https://github.com/ithaka/aiw-ui/blob/f08ae76ec15fb700abbe4b8dd650985bdd8fcc15/.editorconfig#L9

So I went ahead and just fixed this file so that it is 2 everywhere.